### PR TITLE
Separated out the redundant directory check code

### LINKED ISSUE #16: refactor code to be more concise
Any attempts made to refactor redundant code would be highly appreciated :clap:

### PR DESCRIPTION
**Pull request by Cazaimi**
## Description

In order to refactor the code, the change includes separation of the redundant directory check, which checks if `config.json` and `teachcode-solutions` already exists in the pwd, into a new function as it is a discrete operation of its own.

Fixes #16 

## Type of change

- [x] Enhancement 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] [N/A] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] [N/A] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/commands/init.js [hunk 1]
```diff
@@ -88,6 +80,22 @@ const initTasks = async () => {
       )}`,
```
Comment (jamesgeorge007) [inline src/commands/init.js]: You could move [this](https://github.com/madlabsinc/teachcode/pull/37/files#diff-03189079b1b3d77ef73e9e3fdb324d6dL79) part out of the helper method.

File: src/commands/init.js [hunk 2]
```diff
@@ -46,18 +46,10 @@ const showInstructions = kickStart => {
...
  */
-
-const initTasks = async () => {
-  await showBanner(
-    'teachcode',
-    ` Learn to code effectively ${`\t`.repeat(4)} Powered by MadHacks`,
-  );
-  console.log();
-
+const checkRedundantDirectory = () => {
```
Comment (jamesgeorge007) [inline src/commands/init.js]:
```suggestion
const checkIfDirExists = () => {
```

File: src/commands/init.js [hunk 3]
```diff
@@ -88,6 +80,22 @@ const initTasks = async () => {
...
+ */
+
+const initTasks = async () => {
+  await showBanner(
+    'teachcode',
+    ` Learn to code effectively ${`\t`.repeat(4)} Powered by MadHacks`,
+  );
+  console.log();
+
+  checkRedundantDirectory();
```
Comment (jamesgeorge007) [inline src/commands/init.js]:
```suggestion
  checkIfDirExists();
```

**Comment by Cazaimi** [comment]
@jamesgeorge007 , can you give me an idea as to how to pass this commit linter's tests?

**Comment by Cazaimi** [comment]
Also, it might be much more helpful and even provide a cleaner experience if this was a part of the Git pre commit hook like husky.

**Comment by jamesgeorge007** [comment]
The respective update for commit message conventions was brought about by a
recent PR. Pre-commit hook was configured as well, it seems that your
branch might be outdated.

You may check out this guidelines:
https://github.com/madlabsinc/teachcode/blob/next/README.md#how-should-i-write-a-commit-message

**Comment by jamesgeorge007** [comment]
@Cazaimi the respective update for commit message conventions was brought about by a
recent PR. Pre-commit hook was configured as well, it seems that your
branch might be outdated.

You may check out these [guidelines](https://github.com/madlabsinc/teachcode/blob/next/README.md#how-should-i-write-a-commit-message)

**Comment by jamesgeorge007** [comment]
Closing due to inactivity.

**Pull request closed without merging**